pkg/ping/cqlping/cqlping.go

```diff
@@ -33,21 +34,30 @@ var options = []byte{4, 0, 0, 0, 5, 0, 0, 0, 0}
 // receives unexpected payload an error is returned.
 // It returns rtt and error.
 func NativeCQLPing(ctx context.Context, config Config) (rtt time.Duration, err error) {
+	backoff := retry.NewExponentialBackoff(time.Millisecond, config.Timeout, config.Timeout/3, 2, 0)
+	deadline := timeutc.Now().Add(config.Timeout)
 	t := timeutc.Now()
-	defer func() {
-		rtt = timeutc.Since(t)
-		if rtt >= config.Timeout {
-			err = ping.ErrTimeout
+	for attempt := 0; ; attempt++ {
+		err = nativeCQLPingOnce(ctx, config, deadline)
+		if err == nil {
+			return timeutc.Since(t), nil
 		}
-	}()
+		duration := backoff.NextBackOff()
+		if duration == retry.Stop {
+			return timeutc.Since(t), ping.ErrTimeout
+		}
+		time.Sleep(duration)
```

Comment:
Please log the information about retries (attempt variable may be useful then).

Reply:
done